src/modules/dart.rs

```diff
@@ -0,0 +1,80 @@
+use std::process::Command;
+
+use super::{Context, Module, RootModuleConfig, SegmentConfig};
+
+use crate::configs::dart::DartConfig;
+
+/// Creates a module with the current Dart version
+///
+/// Will display the Dart version if any of the following criteria are met:
+///     - Current directory contains a `.dart` file
+///     - Current directory contains a `pubspec.yaml` file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_dart_project = context
+        .try_begin_scan()?
+        .set_files(&["pubspec.yaml"])
+        .set_extensions(&["dart"])
+        .is_match();
+
+    if !is_dart_project {
+        return None;
+    }
+
+    match get_dart_version() {
+        Some(dart_version) => {
+            let mut module = context.new_module("dartlang");
+            let config: DartConfig = DartConfig::try_load(module.config);
+
+            module.set_style(config.style);
+            module.create_segment("symbol", &config.symbol);
+
+            let formatted_version = format_dart_version(&dart_version)?;
+            module.create_segment("version", &config.version.with_value(&formatted_version));
+
+            Some(module)
+        }
+        None => None,
+    }
+}
+
+fn get_dart_version() -> Option<String> {
```

Comment:
This function can also be simplified a bit, the `match` is not necessary here.
```rust
fn get_dart_version() -> Option<String> {
    let output = Command::new("dart").arg("--version").output().ok()?;
    if !output.status.success() {
        log::warn!(
            "Non-Zero exit code '{}' when executing `dart --version`",
            output.status
        );
        return None;
    }

    String::from_utf8(output.stderr).ok()
}
``` 
Another notable change is propagating None instead of panicing when the utf-8 decode fails, so that starship won't break if dart is really screwed up.

Reply:
Thanks for the feedback! Just wondering how would `String::from_utf8(output.stderr).ok()` still propagate `None`? (I'm completely new, to Rust so probably it's newbie question.)

Reply:
whoops, I meant return. It's not propagating here, you are right :)

Reply:
Ah okay, but performance-wise it's the same right?

Reply:
something like this is never gonna affect performance anyway, but the way I suggested should be more efficient unless Rust optimizes it because it only creates one `Option` (`.ok()` does this) while the way I described in my second comment creates two `Option`'s, one with the `.ok()` and another directly from the `Some` call.

Reply:
That's pretty cool thanks!